src/resmom/linux/mom_start.c

```diff
@@ -2242,7 +2242,9 @@ set_globid(job *pjob, struct startjob_rtn *sjr)
 		(void)decode_str(&pjob->ji_wattr[JOB_ATR_acct_id],
 			ATTR_acct_id, NULL, buf);
 
-		*(jid_t *)&pjob->ji_extended.ji_ext.ji_4jid[0] = sjr->sj_jid;
+		(void)memcpy(&pjob->ji_extended.ji_ext.ji_4jid,
+				&sjr->sj_jid,
```

Comment:
@mike0042 what is the size of sj_jid? I see that ji_4jid is 8byte char array and if sj_jid turns out to be anything lesser than 8 bytes then we will end up copying junk

Reply:
This code was changed because it caused a warning related to strict aliasing that can result in undefined behavior at runtime. The new code has well defined behavior that will not copy data past the end of ji_4jid. The two fields are supposed to house the same data, but ji_4jid is defined in the case that we are building for a non-SGI system lacking job.h. At the very least, it doesn't make the situation any worse than what it was.

Reply:
@mike0042 even if both the fields house the same data, it is important that they match in sizes too. If they don't match then memcpy is either going to truncate data or end up copying junk.
I'd suggest we pass smaller of the two sizes while doing memcpy, I guess that is lesser devil in such scenarios.

Reply:
Trying to interpret the data and "do the right thing" is much easier said than done. If they differ in size, do we take the upper or lower bits? It depends on the endianness of the system. While this can be determined, we shouldn't have to care. The point being, they should never differ in size. Eight bytes were used because the jid should always be a 64 bit unsigned integer. I'm not condoning the technique used here to deal with SGI systems, but it has worked for decades. In this case, we're using a function with predictable behavior to silence a compiler warning that is entirely valid.

Reply:
@mike0042 If you are sure that it is 8 bytes then the code is correct. Earlier we were assigning it but now we are copying it and that worries me. I guess endianness should not matter because this information is not something we receive from network, it is probably generated and copied on the same system.
If you have a strong feeling that it is the right thing to do then I'll agree.

Reply:
@arungrover Thanks for your comments. I thought someone might ask about this, since it's really the only "functional" change in the PR. I do feel strongly that the change is correct. While there may be a more elegant way of dealing with the difference between SGI and non-SGI platforms as it relates to jid_t, it would involve much more than this simple workaround. That's not something I want to address in this PR, nor do I think it's anything we need to be overly concerned about.